zookeeper-jute/src/main/java/org/apache/jute/BinaryInputArchive.java

```diff
@@ -85,6 +85,18 @@ public String readString(String tag) throws IOException {
     }
     
     static public final int maxBuffer = Integer.getInteger("jute.maxbuffer", 0xfffff);
```

Comment:
In order to have better testing we can do as follows:
- make maxBuffer and readExtraSize two instance variables
- add two new constants (final static) DEFAULT_MAX_BUFFER and DEFAULT_READEXTRASIZE initialized with the system properties
- in the constructor initialize the value using DEFAULT_MAX_BUFFER and DEFAULT_READEXTRASIZE
- add a new constructor that allows to initialize maxBuffer and readExtraSize
- add only unit tests about BinaryInputArchive, and not end to end tests

Reply:
I liked the approach. thanks.  Addressed the comments more or less same way. Please have a look.  
maxBuffer is used many places and it is not the default value, it is the configured value. So leaving this variable as it is. created other instance variable as suggested but with different name, hope it is ok.